vnext/Scripts/IntegrationTests.ps1

```diff
@@ -1,97 +1,94 @@
+
 	[ValidateSet('x64', 'x86')]
 	[string] $Platform = 'x64',
+
 	[ValidateSet('Debug', 'Release')]
 	[string] $Configuration = 'Debug',
+
 	[string[]] $Tests,
-	$Assembly = "$PSScriptRoot\..\target\$Platform\$Configuration\React.Windows.Desktop.IntegrationTests\React.Windows.Desktop.IntegrationTests.dll",
-	$VsTest = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe",
+
+	[switch] $Preload,
+
 	[UInt32] $Delay = 10,
+
+	[ValidateScript({Test-Path $_})]
+	[string[]] $Assemblies =
+	(
+		"$PSScriptRoot\..\target\$Platform\$Configuration\" +
+		"React.Windows.Desktop.IntegrationTests\React.Windows.Desktop.IntegrationTests.dll"
+	),
+
+	[switch] $NoRun,
+
 	[switch] $List,
-	[switch] $Preload,
-	[string] $ReactNativeHome = "$($PSScriptRoot | Split-Path)\node_modules\react-native"
-)
 
-function Cleanup() {
-	# Expected netstat header:
-	# P1 P2     P3                     P4                     P5              P6
-	#    Proto  Local Address          Foreign Address        State           PID
-	try {
-		$process = (netstat -ano | Select-String ':8081')[0] | ConvertFrom-String
-		(Get-Process -Id $process.p6).Kill()
-	}
-	catch {
-		Write-Host 'Metro bundler process not found.'
-	}
+	[ValidateScript({Test-Path $_})]
+	[string] $VsTest =	"${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\Enterprise\" +
+						"Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe",
 
-	try {
-		$process = (netstat -ano | Select-String ':5555')[0] | ConvertFrom-String
-		(Get-Process -Id $process.p6).Kill()
-	}
-	catch {
-		Write-Host 'WebSocket server process not found.'
-	}
-}
+	[ValidateScript({Test-Path $_})]
+	[string] $ReactNativeLocation = "$($PSScriptRoot | Split-Path)\node_modules\react-native",
+
+	[ValidateScript({Test-Path $_})]
+	[string] $NodePath = (Get-Command node.exe).Definition,
+
+	[ValidateScript({Test-Path $_})]
+	[string] $NpmPath = (Get-Command npm.cmd).Definition
+)
 
-#TODO: Cleanup Watchman logs, free ports 5555 and 8081.
+# Import test server functions.
+. $PSScriptRoot\TestServers.ps1
 
-$procs = @()
+if ($List) {
+	# https://github.com/Microsoft/vstest/issues/1732: Can't print full test names directly to console.
+	& $VsTest ($Assemblies -join ' ') --ListFullyQualifiedTests `
+		--ListTestsTargetPath:$env:TEMP\ReactWindowsIntegrationTestsList.txt
 
-if($List.IsPresent) {
-	Write-Host '-List is present. Skipping setup.'
-	$NoSetup = $true
+	Get-Content $env:TEMP\ReactWindowsIntegrationTestsList.txt
+	Exit
 }
 
-if (! $NoSetup) {
-	$procs += Start-Process powershell -PassThru -WorkingDirectory "$PSScriptRoot\.." -ArgumentList "npm run start"
-	$procs += Start-Process powershell -PassThru -ArgumentList '-NoProfile', "-File $ReactNativeHome\IntegrationTests\Launch-WebSocketServer.ps1 80 24"
+# Ensure test services are online.
+if (! $NoServers) {
+	$packager = Find-Packager
+	$wsServer = Find-WebSocketServer
+	$notFound = $false
```

Comment:
>$notFound [](start = 1, length = 9)

(nit) can we use positive flag, like $packagerFound? all if statements would be easier to read

Reply:
Can't think of a more accurate flag at this point.
This flag doesn't intend to tell whether the servers were found, but whether they should be taken into account.

I.E. if you want to run tests that don't require the servers, you can use the flag to skip both validation and setup.
That said, this should not be the default, hence the switch.